File: UVA 374.cpp
#include<stdio.h>
#include<math.h>
int main()
{
    long double b,p;
    long long ans,m;
    while(scanf("%Lf%Lf%lld",&b,&p,&m)==3)
    {
        ans=0;
        ans=(long long)pow(b,p);
        printf("%lld\n",ans%m);
    }
}

File: UVA 374.java
import java.math.*;
import java.util.*;
public class Main {
	public static void main(String[] args)
	{
		Scanner scan=new Scanner(System.in);
		BigInteger base,md,power,ans;
		while(scan.hasNext())
		{
			ans=new BigInteger("0");
			base=scan.nextBigInteger();
			power=scan.nextBigInteger();
			md=scan.nextBigInteger();
			ans=base.modPow(power, md);
			System.out.println(ans);
		}
	}
}